hudi-common/src/main/java/org/apache/hudi/common/model/HoodieWriteStat.java

```diff
@@ -49,6 +49,12 @@
    */
   private String prevCommit;
 
+  /**
+   * Total number of records written to the previous version of the file slice.
+   * If inflight commit is c2, then number of records present in f1_w1_c1.parquet.
+   */
+  private long oldNumWrites;
```

Comment:
as far as I can tell, we only use this within HoodieMergeHandle.  Can we avoid adding the extra member here and simply use a local variable? I am trying to understand the use-case for logging this in stat.

Reply:
@nbalajee Do you have a reason for this requirement? Seems ok to remove saving it.

Reply:
@vinothchandar - This check can be toggled on/off on a per table basis.  When debugging an actual incident,  where the commit metadata files have been archived, older snapshots of data files have been cleaned up by the cleaner,  having "oldNumWrites" metadata is of great help in identifying the instance that resulted in a smaller parquet file by inspecting the archived commit metadata (Especially when the check is turned off and wouldn't throw an exception).   

For example, if the file were to evolve  from f1_c1.parquet to f1_c10.parquet, then without this "oldNumWrites" information, we have hunt down all the the older archived commits, in search of the commit that hat touched the data file. 

@prashantwason  - in performDatalossCheck(), we should move this out side of the isDatalossCheckEnabled() condition, so that we will record this info, irrespective of whether the flag is enabled or not.

Reply:
@nbalajee If we move this outside of  isDatalossCheckEnabled() condition, then wont it always read the older parquet files each time before commit? If so, there is not much to the setting anyways because the overhead of this feature is reading the older parquet each time before merge.

Reply:
@nbalajee I totally understand. but we could log this out and use elastic search or some other infrastructure to debug right. I would really love to not mix debug information with something as critical as commit metadata. 
This sets a precedent that we cannot follow going forward :)